pymongo/change_stream.py

```diff
@@ -283,12 +298,15 @@ def try_next(self):
         # one resume attempt.
         try:
             change = self._cursor._try_next(True)
-        except ConnectionFailure:
+        except (ConnectionFailure, CursorNotFound):
             self._resume()
             change = self._cursor._try_next(False)
         except OperationFailure as exc:
-            if (exc.code in _NON_RESUMABLE_GETMORE_ERRORS or
-                    exc.has_error_label("NonResumableChangeStreamError")):
+            is_resumable = ((exc._max_wire_version >= 9 and
```

Comment:
Is there any chance that `_max_wire_version` could be None here?

Reply:
AFAICT this is not possible. The only place we seem to be raising `OperationFailure` in this code path is via _check_command_response which always populates `_max_wire_version`

Reply:
I don't think this is necessarily the case. Auth failures might raise OperationFailure without  `_check_command_response` for example: https://github.com/mongodb/mongo-python-driver/blob/e608ff41662c694f7ba84e9d1a004463677886cf/pymongo/auth.py#L301

Can we guard against the None case?